File: app/src/main/java/tech/pauly/findapet/data/PetfinderEndpoints.java
package tech.pauly.findapet.data;

import tech.pauly.findapet.BuildConfig;

public class PetfinderEndpoints {
    public static String getEndpoint() {
        String environment = BuildConfig.ENVIRONMENT;
        switch (environment) {
            case "prod":
                return "http://api.petfinder.com/";
            case "espresso":
                return "http://localhost:8010/";
            default:
                throw new IllegalStateException("PetfinderEndpoints " + environment + " is not a supported environment");
        }
    }
}

File: app/src/main/java/tech/pauly/findapet/data/PetfinderEndpoints.kt
package tech.pauly.findapet.data

import tech.pauly.findapet.BuildConfig

object PetfinderEndpoints {
    val endpoint: String
        get() {
            val environment = BuildConfig.ENVIRONMENT
            return when (environment) {
                "prod" -> "http://api.petfinder.com/"
                "espresso" -> "http://localhost:8010/"
                else -> throw IllegalStateException("PetfinderEndpoints $environment is not a supported environment")
            }
        }
}
